calyx/src/ir/from_ast.rs

```diff
@@ -44,9 +44,16 @@ pub fn ast_to_ir(
 
     // Add component signatures
     for comp in &components {
-        sig_ctx
-            .comp_sigs
-            .insert(comp.name.clone(), comp.signature.clone());
+        let mut sig = comp.signature.clone();
+        sig.inputs.push(ast::Portdef {
+            name: "go".into(),
+            width: 1,
+        });
+        sig.outputs.push(ast::Portdef {
+            name: "done".into(),
+            width: 1,
+        });
+        sig_ctx.comp_sigs.insert(comp.name.clone(), sig);
```

Comment:
This is incorrect. See the `Component::new` function to see the names of all the ports generated. I would recommend creating a `pub(super)` method that returns the extended port list and use it in both `Component::new` and this function.